app/models/user.rb

```diff
@@ -92,4 +105,26 @@ def ensure_no_token_scope_loss
   def ensure_last_active_at_presence
     self.last_active_at ||= Time.zone.now
   end
+
+  # Internal: Returns a SimpleBox encryptor for token encryption/decryption.
+  def encryptor
+    token_secret = Rails.application.secrets.token_secret
+    raise "TOKEN_SECRET is not set, please check you .env file" if token_secret.blank?
+    @encryptor ||= RbNaCl::SimpleBox.from_secret_key(token_secret.unpack("m0").first)
+  end
+
+  def encrypt_value(value)
+    return if value.blank?
+
+    ciphertext = encryptor.encrypt(value)
+    hex = ciphertext.bytes.pack("c*").unpack("H*").first
+    "GH_" + hex
+  end
+
+  def decrypt_value(value)
+    return value unless value && value =~ /\AGH_/
+
+    hex = value.sub(/\AGH_/, '')
```

Comment:
If using base64 above this would become:

```
  def decrypt_value(value)
    return value unless value && value =~ /\AGH_/
    ciphertext = value.sub(/\AGH_/, '')
    encryptor.decrypt(Base64.decode64(ciphertext))
  end
```